### ISSUE TITLE
[Bug] Date picker: February displays incomplete set of days

### ISSUE THREAD

**Issue by sallaben** [illright/attractions#243]
This is an interesting one -- I am looking into it myself, but I thought you might want to know as well: February does not display correctly in the attractions datepicker.

Reproduced in the docs: https://illright.github.io/attractions/docs/components/date-picker

![image](https://user-images.githubusercontent.com/7699235/107886022-a966ef80-6eba-11eb-8596-8fa446bd52ca.png)

**Comment by illright** [comment]
Cannot reproduce on desktop/mobile Chrome and on Firefox. Could you tell me what's your browser and system locale?

**Comment by sallaben** [comment]
Mozilla/5.0, Gecko/20100101 Firefox/85.0 -- for what it's worth, I can reproduce on mobile (Safari). Maybe it's a locale based bug?

**Comment by aabounegm** [comment]
I also couldn't reproduce on Firefox 84 or 85.0.2, and my locale is also set to `en-US`. I have no access to Safari to test it there, though.

**Comment by sallaben** [comment]
Very interesting, thanks for the comments guys... I looked deeper into the calendar component and datetime utils, and the issue boils down to the setMonth functionality on JS Date:

```
  const foo = new Date(0);
  foo.setFullYear(2021, 1); // February 2021 (setMonth is used by setFullYear implementation as well)
  console.log(foo);
```

We expect this return a Date object for February 1st, and when I run this code in a UTC node container, I get the expected result: `2021-02-01T00:00:00.000Z`. But when I run the exact same code locally in node (15.8.0) or in my browser, the result is `2021-03-04T00:00:00.000Z`. I'm thinking maybe the way we create the day cursor, using new Date(0), is failing in non UTC environments (I am in Mountain Standard Time).

Here's the gist, from what I understand. When the date is, for example, May 31 2021, and we set the month to February, it becomes February 31 2021 -- which then becomes March 3 2021, since there is no February 31.
- Here are two StackOverflow questions about this topic. [1](https://stackoverflow.com/questions/51581452/wrong-javascript-month-after-2020) [2](https://stackoverflow.com/questions/23395934/javascript-date-bug-february-2014)
- I have identified a fix for this behavior. When using setMonth or setFullYear to change the month value, ensure that we also pass a date value in at the same time.

Here is a PR to fix the behavior in the datetime-utils: #244  

And February now displays normally :-) 
![image](https://user-images.githubusercontent.com/7699235/107890472-8c401a00-6ed6-11eb-986a-e3415f11c124.png)

**Comment by aabounegm** [comment]
It was confusing to me how these links are relevant when the date today is the 15th (or 14th for you), and especially that `new Date(0)` should return the epoch (`1970.01.01`), irrelevant of the current timezone (I was wrong!). Turns out that the timezone actually affects even that constructor, so I tried changing my timezone to **US/Pacific** (the one available in DevTools) and running `new Date(0)`, and it indeed gave me `1969.12.31`!
Thank you for pointing this out, I'll review the PR now.

_P.S.: I can get used to `MM.DD.YYYY`, but I've never seen a locale where the week starts on Friday!_
